framework/assets/yii.js

```diff
@@ -143,74 +144,70 @@ window.yii = (function ($) {
+                pjax = $e.data('pjax'),
+                usePjax = pjax !== undefined && pjax !== 0 && $.support.pjax,
                 pjaxContainer,
                 pjaxOptions = {};
 
             if (usePjax) {
-                if ($e.data('pjax-container')) {
-                    pjaxContainer = $e.data('pjax-container');
-                } else {
-                    pjaxContainer = $e.closest('[data-pjax-container=""]');
-                }
-                // default to body if pjax container not found
+                pjaxContainer = $e.data('pjax-container') || $e.closest('[data-pjax-container]');
                 if (!pjaxContainer.length) {
                     pjaxContainer = $('body');
                 }
                 pjaxOptions = {
                     container: pjaxContainer,
-                    push: pjaxPushState,
-                    replace: pjaxReplaceState,
-                    scrollTo: pjaxScrollTo,
-                    pushRedirect: pjaxPushRedirect,
-                    replaceRedirect: pjaxReplaceRedirect,
-                    pjaxSkipOuterContainers: pjaxSkipOuterContainers,
-                    timeout: pjaxTimeout,
+                    push: !!$e.data('pjax-push-state'),
+                    replace: !!$e.data('pjax-replace-state'),
+                    scrollTo: $e.data('pjax-scrollto'),
+                    pushRedirect: $e.data('pjax-push-redirect'),
+                    replaceRedirect: $e.data('pjax-replace-redirect'),
+                    skipOuterContainers: $e.data('pjax-skip-outer-containers'),
+                    timeout: $e.data('pjax-timeout'),
                     originalEvent: event,
                     originalTarget: $e
-                }
+                };
             }
 
             if (method === undefined) {
-                if (action && action != '#') {
-                    if (usePjax) {
-                        $.pjax.click(event, pjaxOptions);
-                    } else {
-                        window.location = action;
-                    }
+                if (isValidAction) {
+                    usePjax ? $.pjax.click(event, pjaxOptions) : window.location.assign(action);
                 } else if ($e.is(':submit') && $form.length) {
                     if (usePjax) {
-                        $form.on('submit',function(e){
+                        $form.on('submit', function (e) {
                             $.pjax.submit(e, pjaxOptions);
-                        })
+                        });
                     }
                     $form.trigger('submit');
                 }
                 return;
             }
 
-            var newForm = !$form.length;
-            if (newForm) {
-                if (!action || !/(^\/|:\/\/)/.test(action)) {
-                    action = window.location.href;
+            var oldMethod,
+                oldAction,
+                newForm = !$form.length;
+            if (!newForm) {
```

Comment:
- It's better to have shorter part of `if` first so it's easier to read.
- Have put the part when the form is not exist under `if` check, We do not need any `oldMethod` / `oldAction` processing in case of existing form.